web/third/bootstrap-typeahead/typeahead.js

```diff
@@ -564,7 +564,7 @@ import {get_string_diff} from "../../src/util";
                         e.keyCode === 8 &&
                         this.options.helpOnEmptyStrings && // Support for inputs to set the hideOnEmpty option explicitly to false
                         // to display typeahead after hitting backspace to clear the input.
-                        (typeof this.options.hideOnEmpty === undefined || this.options.hideOnEmpty)
+                        (this.options.hideOnEmpty === undefined || this.options.hideOnEmpty)
```

Comment:
The old code was a bug. If the option is missing, `typeof this.options.hideOnEmpty` would have been the string `"undefined"`, not `undefined`. So this change clearly aligns with what the author of #25135 *meant* to write. But the question is, are we relying on this bug?

Reply:
Hm, so it sounds like the actual functionally equivalent thing to do would be to remove the undefined equality check completely. So maybe we can do that for now, to keep it working as is, and we can add some undefined check back if we realize that's what we want instead?

Reply:
Updated to that for now, with more detail in the commit message.

Reply:
Looks like `hideOnEmpty` actually isn't being used anywhere anymore

Reply:
I think `hideOnEmpty` was implement as part of taming the typeahead for the previous attempt at search pills. Seems fine to remove and we can re-add something like if it needed for the new search pills UI.

Reply:
I've updated the commit to remove it instead.